### PR TITLE
Feat/worklogs

### PR DESCRIPTION
**Pull request by sanderdejong88**

### CODE REVIEW

File: api/bridge/worklogs/[worklogId].ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    const {body} = request;
+    const timeSpent = body.record.hours * 60 * 60;
+    const worklog = {
+      time_spent: timeSpent,
+      time_billable: timeSpent,
+      start_time: format(new Date(body.record.date), 'yyyy-MM-dd HH:mm:ss'),
+      contract_id: body.contractId,
+      description: 'synced from on-site hours',
+      source: 'on-site',
+      source_id: 1, // random number, we don't need an identifier
```
Comment (sanderdejong88) [inline api/bridge/worklogs/[worklogId].ts:19]: Source id should be updated with the put

File: api/bridge/worklogs/[worklogId].ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+export default async function Worklogs(request: VercelRequest, response: VercelResponse) {
+  const worklogId = request.query.worklogId;
+
+  if (request.method === 'PUT') {
+    const {body} = request;
+    const timeSpent = body.record.hours * 60 * 60;
+    const worklog = {
+      time_spent: timeSpent,
+      time_billable: timeSpent,
+      start_time: format(new Date(body.record.date), 'yyyy-MM-dd HH:mm:ss'),
```
Comment (sanderdejong88) [inline api/bridge/worklogs/[worklogId].ts:15]: Use 8AM

File: api/bridge/worklogs/index.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    source_id: new Date().getTime(), // unique number, we don't need an identifier
+    source_reference: '',
+  };
+
+  try {
+    const {
+      data: {
+        data: {id: worklogId},
+      },
+    } = await axios.post(`${process.env.BRIDGE_URL}/api/v1/worklogs`, worklog, {
```
Comment (FrankMerema) [inline api/bridge/worklogs/index.ts:27]: Usually I add a .then(({data}) => data.data); So the const can be simplified :)

File: components/records/weekly-timesheet-footer.vue [hunk 4]
```diff
@@ -50,28 +50,47 @@
...
-          {{$t('undo')}}
-        </b-button>
+        <div v-if="isApproved">
+          <b-button-group class="mr-3">
+            <b-button
+              variant="outline-primary"
+              :disabled="isSaving"
+              @click="handleRemoveFromBridgeClick"
+            >
+              Remove from bridge
```
Comment (FrankMerema) [inline components/records/weekly-timesheet-footer.vue:73]: Not using the translating here ;)

File: components/records/weekly-timesheet-footer.vue [hunk 5]
```diff
@@ -50,28 +50,47 @@
...
+            </b-button>
+
+            <b-button
+              variant="outline-primary"
+              :disabled="isSaving"
+              @click="handleAddToBridgeClick"
+            >
+              Add to bridge
```
Comment (FrankMerema) [inline components/records/weekly-timesheet-footer.vue:81]: Not using the translating here ;)

File: store/auth/actions.ts [hunk 6]
```diff
@@ -44,6 +44,7 @@ const actions: ActionTree<AuthStoreState, RootStoreState> = {
     } catch {
       const authState = await dispatch('logout');
       if (authState) {
+        // @ts-ignore
```
Comment (FrankMerema) [inline store/auth/actions.ts:47]: Why ignoring this line?

**Review by FrankMerema — COMMENTED** [review]
Couple of minor things ;)

File: store/auth/actions.ts (hunk 6 above)
Comment (sanderdejong88) [reply store/auth/actions.ts:47]: i18n module isn't adding the types correctly and it annoyed me to get the error often...

**Review by FrankMerema — APPROVED** [review]

**Pull request merged**